src/Validation/ValidationRule.php

```diff
@@ -165,54 +100,19 @@ public function process(mixed $value, array $providers, array $context = []): ar
      */
     protected function _skip(array $context): bool
     {
-        if (is_string($this->_on)) {
+        if (is_string($this->on)) {
             $newRecord = $context['newRecord'];
 
-            return ($this->_on === Validator::WHEN_CREATE && !$newRecord)
-                || ($this->_on === Validator::WHEN_UPDATE && $newRecord);
+            return ($this->on === Validator::WHEN_CREATE && !$newRecord)
+                || ($this->on === Validator::WHEN_UPDATE && $newRecord);
         }
 
-        if ($this->_on !== null) {
-            $function = $this->_on;
+        if ($this->on !== null) {
+            $function = $this->on;
 
             return !$function($context);
         }
 
         return false;
     }
-
-    /**
-     * Sets the rule properties from the rule entry in validate
-     *
-     * @param array<string, mixed> $validator [optional]
-     * @return void
-     */
-    protected function _addValidatorProps(array $validator = []): void
-    {
-        foreach ($validator as $key => $value) {
-            if (!$value) {
-                continue;
-            }
-            if ($key === 'rule' && is_array($value) && !is_callable($value)) {
-                $this->_pass = array_slice($value, 1);
-                $value = array_shift($value);
-            }
-            if (in_array($key, ['rule', 'on', 'message', 'last', 'provider', 'pass'], true)) {
-                $this->{"_{$key}"} = $value;
-            }
-        }
-    }
-
-    /**
-     * Returns the value of a property by name
-     *
-     * @param string $property The name of the property to retrieve.
-     * @return mixed
-     */
-    public function get(string $property): mixed
-    {
-        $property = '_' . $property;
-
-        return $this->{$property} ?? null;
-    }
```

Comment:
How do we deprecate this? I can't think of a way to do it that doesn't require property hooks. Another simple solution is to retain this method with deprecation warnings in 6.x. This is a method call we could rename with rector as well.

Reply:
We could retain it with deprecation warning in 6.x, but is it really necessary? Users don't need to interact with this class directly. I never knew about their existence until I dug into how the validation process works :) These methods were only used by the validator class.

Reply:
Ok. Maybe this method we don't need a shim for :smile: